### PR TITLE
#36 Use storage cell for SparePart

### PR DESCRIPTION
**Pull request by a-v-dubr**

### CODE REVIEW

**Review by dmytroshchotkin — APPROVED** [review]

File: PartsApp/DatabaseHelper/DbStructureCheck.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            using (var connection = DbConnectionHelper.GetDatabaseConnection() as SQLiteConnection)
+            {
+                connection.Open();
+                EnsureStorageCellColumnExistsInSparePartsDB(new SQLiteCommand("", connection));            
+                connection.Close();
+            }
+        }
+
+        private static void EnsureStorageCellColumnExistsInSparePartsDB(SQLiteCommand cmd)
```
Comment (dmytroshchotkin) [inline PartsApp/DatabaseHelper/DbStructureCheck.cs]: We have one DB only

File: PartsApp/DatabaseHelper/DbStructureCheck.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+namespace PartsApp.DatabaseHelper
+{
+    internal static class DbStructureCheck
+    {
+        internal static void EnsureNewDBColumnsExists()
+        {
+            using (var connection = DbConnectionHelper.GetDatabaseConnection() as SQLiteConnection)
+            {
+                connection.Open();
+                EnsureStorageCellColumnExistsInSparePartsDB(new SQLiteCommand("", connection));
```
Comment (dmytroshchotkin) [inline PartsApp/DatabaseHelper/DbStructureCheck.cs]: Why pass a dummy CMD object as a parameter? And conventionally also this is incorrect: typically we pass an open connection

File: PartsApp/DatabaseHelper/DbStructureCheck.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+﻿using Infrastructure;
+using System;
+using System.Data.SQLite;
+
+namespace PartsApp.DatabaseHelper
+{
+    internal static class DbStructureCheck
```
Comment (dmytroshchotkin) [inline PartsApp/DatabaseHelper/DbStructureCheck.cs]: schema

File: PartsApp/Form1.cs [hunk 4]
```diff
@@ -1168,7 +1166,7 @@ private void Form1_FormClosed(object sender, FormClosedEventArgs e)
         {
             //Если Титор, тогда делаем бэкап (Для проверки на Lock conflict)
             if (CurEmployee?.EmployeeId == 1)
```
Comment (dmytroshchotkin) [inline PartsApp/Form1.cs:1168]: This is a bad approach. Create a US somewhere for refactoring to change this and to come up with another backup mechanism.

**Review by dmytroshchotkin — APPROVED** [review]

File: PartsApp/DatabaseHelper/DbStructureCheck.cs (hunk 1 above)
Comment (a-v-dubr) [reply PartsApp/DatabaseHelper/DbStructureCheck.cs]: edited

File: PartsApp/DatabaseHelper/DbStructureCheck.cs (hunk 2 above)
Comment (a-v-dubr) [reply PartsApp/DatabaseHelper/DbStructureCheck.cs]: corrected

File: PartsApp/DatabaseHelper/DbStructureCheck.cs (hunk 3 above)
Comment (a-v-dubr) [reply PartsApp/DatabaseHelper/DbStructureCheck.cs]: edited

File: PartsApp/Form1.cs (hunk 4 above)
Comment (a-v-dubr) [reply PartsApp/Form1.cs:1168]: done
https://dev.azure.com/razor23donetsk0618/TruckTir/_workitems/edit/41

**Review by dmytroshchotkin — APPROVED** [review]

**Pull request merged**